server/src/modules/refresh-tokens/refresh-tokens.service.ts

```diff
@@ -0,0 +1,46 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { RefreshTokenEntity } from './refresh-token.entity';
+import { refreshTokensServiceErrorMessages } from './refresh-tokens.constants';
+import { BaseService } from 'src/common/services';
+import * as ms from 'ms';
+import { AppConfigService } from 'src/config/app-config.service';
+import { UsersService } from '../users/users.service';
+
+@Injectable()
+export class RefreshTokensService extends BaseService<RefreshTokenEntity> {
+  constructor(
+    @InjectRepository(RefreshTokenEntity)
+    private readonly refreshTokenEntityRepository: Repository<RefreshTokenEntity>,
+    private readonly appConfigService: AppConfigService,
+    private readonly usersService: UsersService,
+  ) {
+    super(refreshTokenEntityRepository, refreshTokensServiceErrorMessages);
+  }
+
+  async updateRefreshToken(userId: string, refreshToken: string) {
+    const user = await this.usersService.findOne({ id: userId });
+
+    const tokenLifetime = this.appConfigService.get('JWT_REFRESH_EXPIRES_IN');
+    const tokenDuration = ms(tokenLifetime);
+    const currentDateTime = new Date();
+    const expiresAt = new Date(currentDateTime.getTime() + tokenDuration);
+
+    const refreshTokenModel: Partial<RefreshTokenEntity> = {
+      user,
+      value: refreshToken,
+      expiresAt,
+    };
+
```

Comment:
I suggest, instead of updating current existing user token, we can simply create new token (for using more than 1 device with different refresh tokens)

And also we can add calling method to remove old user refresh tokens from db (to keep, for example, only 5 active refresh tokens):
await deleteOldRefreshTokens({id: user.id})

---> select all user refresh tokens, and delete refresh tokens which were created after first, for example 3, tokens (using createdAt field)

Reply:
for example, await this.deleteAll({ where: {id: user.id}, orderBy: { createdAt: 'DESC' }, skip: 3 })